### ISSUE TITLE
Add operation tracking to VmStateIterator

### ISSUE THREAD

**Issue by bobbinth** [maticnetwork/miden#235]
Currently, the [VmState](https://github.com/maticnetwork/miden/blob/next/processor/src/debug.rs#L7) struct which is returned from the `VmStateIterator` does not contain the operation which was executed to put the VM into this state. We should add another field to this struct so that it looks something like this:
```Rust
#[derive(Clone, Debug, Eq, PartialEq)]
pub struct VmState {
    pub op: Operation,
    pub clk: usize,
    pub fmp: Felt,
    pub stack: Vec<Felt>,
    pub memory: Vec<(u64, Word)>,
}
```

The `op` field would need to be populated from the information in the [decoder](https://github.com/maticnetwork/miden/blob/next/processor/src/decoder/mod.rs#L209). The decoder doesn't explicitly track the operations yet. It is possible to infer the operations from the trace, but I think a better approach would be to have a vector of operations in the decoder struct. Then, as the VM executes operations, they would be pushed into this vector.

Since tracking operations would result in some overhead, the above should happen only when we are executing programs via [execute_iter()](https://github.com/maticnetwork/miden/blob/next/processor/src/lib.rs#L80) function.

Implementing this functionality will enable counting operations executed by the VM which would be useful for things like #198.

**Comment by tohrnii** [comment]
@bobbinth I can complete this first as one PR and use that to complete #198.

**Comment by bobbinth** [comment]
That would be awesome! Thank you!

**Comment by bobbinth** [comment]
Closed by #251
